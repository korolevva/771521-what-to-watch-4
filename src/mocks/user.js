const user = {
  avatarUrl: `mock/user`,
  email: `foo2google.com`,
  id: 1,
  name: `mockUser`,
};

export default user;
