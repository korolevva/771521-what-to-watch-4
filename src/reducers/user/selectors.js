import NameSpace from "../name-space.js";

export const getAuthorizationStatus = (store) => {
  return store[NameSpace.USER].authorizationStatus;
};

export const getIsSingInSelected = (store) => {
  return store[NameSpace.USER].isSingInSelected;
};

export const getIsErrorAuth = (store) => {
  return store[NameSpace.USER].isErrorAuth;
};

export const getUserInfo = (state) => {
  return state[NameSpace.USER].userInfo;
};
