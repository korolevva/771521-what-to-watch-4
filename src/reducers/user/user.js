import {ActionType} from "../../const.js";

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isSingInSelected: false,
  isErrorAuth: false,
};

export const userReduser = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return Object.assign({}, state, {
        authorizationStatus: action.payload,
      });
    case ActionType.RENDER_SIGN_IN_PAGE:
      return Object.assign({}, state, {
        isSingInSelected: action.payload,
      });
    case ActionType.RENDER_MAIN_PAGE:
      return Object.assign({}, state, {
        isSingInSelected: action.payload,
      });
    case ActionType.CHECK_ERROR_AUTHORIZATION:
      return Object.assign({}, state, {
        isErrorAuth: action.payload,
      });
    default:
      return state;
  }
};

