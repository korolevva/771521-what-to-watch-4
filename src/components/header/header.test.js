import React from 'react';
import renderer from 'react-test-renderer';
import Header from './header.jsx';
import {Router} from "react-router-dom";
import history from "../../history.js";
import user from "../../mocks/user.js";

describe(`Render Header`, () => {
  it(`Render MovieCard with avatar`, () => {
    const tree = renderer.create(
        <Router
          history={history}
        >
          <Header
            authorizationStatus={`AUTH`}
            user={user}
          />
        </Router>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieCard without avatar`, () => {
    const tree = renderer.create(
        <Router
          history={history}
        >
          <Header
            authorizationStatus={`NO_AUTH`}
            user={user}
          />,
        </Router>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
