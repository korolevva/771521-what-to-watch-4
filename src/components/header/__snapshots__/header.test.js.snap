// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Render MovieCard with avatar 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      href="/my-list"
      onClick={[Function]}
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="mock/user"
          width="63"
        />
      </div>
    </a>
  </div>
</header>
`;

exports[`Render Header Render MovieCard without avatar 1`] = `
Array [
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>,
  ",",
]
`;
